arangod/Utils/SupportInfoBuilder.cpp

```diff
@@ -0,0 +1,762 @@
+                               LogicalView::ptr const& view) -> bool {
+                             dbViews[database]++;
+
+                             return true;
+                           });
+  }
+
+  struct DbCollStats {
+    size_t numDocColls;
+    size_t numGraphColls;
+    size_t numSmartColls;
+    size_t numDisjointGraphs;
+    VPackBuilder builder;
+  };
+
+  std::unordered_map<std::string_view, DbCollStats> visitedDatabases;
+  containers::FlatHashMap<size_t, containers::FlatHashSet<std::string_view>>
+      visitedColls;
+  containers::FlatHashMap<size_t, size_t> collNumDocs;
+  // merge all collections belonging to the database into
+  // the same object, as the database might be in more
+  // than one dbserver
+  for (auto const dbItFromServers : velocypack::ObjectIterator(infoSlice)) {
+    for (auto const dbIt : velocypack::ArrayIterator(dbItFromServers.value)) {
+      std::string_view dbName = dbIt.get("name").stringView();
+      if (visitedDatabases.find(dbName) == visitedDatabases.end()) {
+        visitedDatabases.insert(std::make_pair(dbName, DbCollStats()));
+        visitedDatabases[dbName].builder.openObject();
+        visitedDatabases[dbName].builder.add("n_views",
+                                             VPackValue(dbViews[dbName]));
+        visitedDatabases[dbName].builder.add(
+            "single_shard", VPackValue(dbIt.get("single_shard").getBoolean()));
+        visitedDatabases[dbName].builder.add("colls",
+                                             VPackValue(VPackValueType::Array));
+      }
+
+      for (auto const collIt : velocypack::ArrayIterator(dbIt.get("colls"))) {
+        size_t planId = collIt.get("plan_id").getNumber<decltype(planId)>();
+
+        std::string_view collName = collIt.get("name").stringView();
+        if (auto const visitedCollIt = visitedColls.find(planId);
+            visitedCollIt != visitedColls.end()) {
+          if (visitedCollIt->second.find(collName) ==
+              visitedCollIt->second.end()) {
+            auto amountDocs = collNumDocs[planId];
+            collNumDocs[planId] +=
+                collIt.get("n_docs").getNumber<decltype(amountDocs)>();
+            visitedColls[planId].insert(collName);
+          }
+        } else {
+          auto amountDocs = collNumDocs[planId];
+          collNumDocs[planId] +=
+              collIt.get("n_docs").getNumber<decltype(amountDocs)>();
+          visitedDatabases[dbName].builder.add(collIt);
+          std::string_view collType = collIt.get("type").stringView();
+          if (collType == "document") {
+            visitedDatabases[dbName].numDocColls++;
+          } else if (collType == "edge") {
+            visitedDatabases[dbName].numGraphColls++;
+          }
+          if (collIt.get("smart_graph").getBoolean()) {
+            visitedDatabases[dbName].numSmartColls++;
+          }
+          if (collIt.get("disjoint").getBoolean()) {
+            visitedDatabases[dbName].numDisjointGraphs++;
```

Comment:
Isn't it the number of disjoint SmartGraph edge collections rather than the number of disjoint graphs that we are counting here?
The member name in the statistics object (and also in the result that we build up) should IMO reflect that.